project-base/storefront/components/Blocks/Banners/DeferredBannersSlider.tsx

```diff
@@ -0,0 +1,37 @@
+import { useSliderItemsQuery } from 'graphql/requests/sliderItems/queries/SliderItemsQuery.generated';
+import { useDeferredRender } from 'hooks/helpers/useDeferredRender';
+import dynamic from 'next/dynamic';
+import { useSessionStore } from 'store/useSessionStore';
+
+const BannersSlider = dynamic(() => import('./BannersSlider').then((component) => component.BannersSlider), {
+    ssr: false,
+    loading: () => <BannersSliderPlaceholder />,
+});
+
+const BannersSliderPlaceholder = dynamic(() =>
+    import('./BannersSliderPlaceholder').then((component) => component.BannersSliderPlaceholder),
+);
+
+const SkeletonModuleBanners = dynamic(
+    () =>
+        import('components/Blocks/Skeleton/SkeletonModuleBanners').then((component) => component.SkeletonModuleBanners),
+    {
+        ssr: false,
+    },
+);
+
+export const DeferredBannersSlider: FC = () => {
+    const [{ data: sliderItemsData, fetching }] = useSliderItemsQuery();
+    const hadClientSideNavigation = useSessionStore((s) => s.hadClientSideNavigation);
+    const shouldRender = useDeferredRender('banners', hadClientSideNavigation);
+
+    if (fetching) {
+        return <SkeletonModuleBanners />;
+    }
+
+    if (!sliderItemsData?.sliderItems.length) {
+        return null;
+    }
+
+    return shouldRender ? <BannersSlider sliderItems={sliderItemsData.sliderItems} /> : <BannersSliderPlaceholder />;
```

Comment:
I am not 100% sure how it will work but wouldn't be easier to have implemented `React.lazy()` with `Suspense`. I am not sure if you took a look on this feature, but with `fallback` it should do the same thing. https://react.dev/reference/react/lazy#suspense-for-code-splitting

Reply:
I worked with this directly
<img width="606" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/c3c1605a-2e96-44c8-9357-6baa8f9eaf93">
https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading

In next.js, it should do the exact same thing. However, the entire problem is that, at least from what I have noticed. If I wrapped 10 things in 10 different suspense boundaries, they were all still hydrated at the same time, in the same task. So it did not help with task splitting and TBT.